cylc/flow/scheduler.py

```diff
@@ -737,7 +737,7 @@ def restart_remote_init(self):
             itask.platform['install target'] = (
                 get_install_target_from_platform(itask.platform))
             if (
-                itask.state(*TASK_STATUSES_ACTIVE)
+                itask.state(*TASK_STATUSES_ACTIVE, TASK_STATUS_PREPARING)
```

Comment:
Without this remote-init on restart could be skipped for preparing tasks?

Reply:
Tasks can't be `preparing` here. On restart, we immediately reset `preparing` tasks to `waiting` so that they re-prep from scratch (with a new submit number).

Reply:
A comment on this in the code would be good though, for future reference.  (I just had to do some quick testing to confirm the behaviour).

Reply:
Added a comment and an integration test.